addons/mail/models/mail_channel_member.py

```diff
@@ -73,11 +73,16 @@ def create(self, vals_list):
         It can not be implemented in XML, because when the record will be created, the
         partner will be added in the channel and the security rule will always authorize
         the creation.
+
+        - ``mail_create_noinvite_check``: at create, do not check the rights of the partners
+        to be invited into the channel.
         """
         if not self.env.is_admin():
             for vals in vals_list:
                 if 'channel_id' in vals:
                     channel_id = self.env['mail.channel'].browse(vals['channel_id'])
+                    if self._context.get('mail_create_noinvite_check'):
```

Comment:
"string" context is not safe and cannot be used for security purposes (it can be given by rpc)

if you absolutely need a context key, it should be a value that cannot be provided by rpc, such as the reference to a technical object that is compared to expected object

Reply:
I tried to directly link the channel members into the channel when creating. is it a better way to do so?